File: 2022/zig/day1/src/day1p2.zig
const std = @import("std");
const lib = @import("lib.zig");
const stdout = std.io.getStdOut().writer();

pub fn main() !void {
    var list = try lib.get_elf_calories("input");

    if (list.items.len >= 3) {
        const rc = list.items[0] + list.items[1] + list.items[2];
        try stdout.print("{d}", .{rc});
    }

    list.deinit();
}

File: 2022/rust/day1/src/day1p2.rs
use lib;

fn main() {
    let elf_cal = lib::get_elf_calories("input"); 

    if elf_cal.len() > 3 {
        let s = &elf_cal[0..3];
        let amt = s.iter().fold(0, |acc, x| acc + x);
        println!("{}", amt);
    } else {
        println!("Error: No elf calories");
    }
}

